python/ray/_common/network_utils.py

```diff
@@ -28,6 +35,50 @@ def build_address(host: str, port: Union[int, str]) -> str:
     return _build_address(host, port)
 
 
+def node_ip_address_from_perspective(address: Optional[str] = None) -> str:
+    """IP address by which the local node can be reached *from* the `address`.
+
+    If no address is given, defaults to public DNS servers for detection.
+
+    Args:
+        address: The IP address and port of any known live service on the
+            network you care about.
+
+    Returns:
+        The IP address by which the local node can be reached from the address.
+    """
+    return _node_ip_address_from_perspective(address)
+
+
+def is_ipv6(address: str) -> bool:
```

Comment:
```suggestion
def is_ipv6(host: str) -> bool:
```

Per the BuildAddress naming convention, address includes port